OpenRA.Mods.Common/Traits/Render/ProductionBar.cs

```diff
@@ -77,7 +77,12 @@ void ITick.Tick(Actor self)
 				return;
 
 			var current = queue.AllQueued().Where(i => i.Started).MinByOrDefault(i => i.RemainingTime);
-			value = current != null ? 1 - (float)current.RemainingCost / current.TotalCost : 0;
+			if (current == null)
+				value = 0;
+			else if (current.TotalCost <= 0)
+				value = 1;
```

Comment:
shouldn't this be 0 as well?

Reply:
No, the unit will be produced instantly so we want to display a full bar.
> Setting the value to one displays the bar correctly as full for the split second the unit is build.